### ISSUE TITLE
Psalm causing ParseError

### ISSUE THREAD

**Issue by jrushlow** [vimeo/psalm#3089]
Good morning, 

I ran `composer update` this morning and using `Installing vimeo/psalm (dev-master 067104e)` Psalm causes a `ParseError` that is uncaught. It I believe this is also appearing in your CI. 

If there is any additional information needed, please let me know.

Stacktrace:

```
developer@71447f9a1646:/var/htdocs$ vendor/bin/psalm
Scanning files...
Uncaught ParseError: syntax error, unexpected ')' in /var/htdocs/vendor/vimeo/psalm/src/Psalm/Type.php:1772
Stack trace:
#0 /var/htdocs/vendor/composer/ClassLoader.php(322): Composer\Autoload\includeFile('/var/htdocs/ven...')
#1 [internal function]: Composer\Autoload\ClassLoader->loadClass('Psalm\\Type')
#2 /var/htdocs/vendor/vimeo/psalm/src/Psalm/Internal/Codebase/CallMap.php(237): spl_autoload_call('Psalm\\Type')
#3 /var/htdocs/vendor/vimeo/psalm/src/Psalm/Internal/PhpVisitor/ReflectorVisitor.php(789): Psalm\Internal\Codebase\CallMap::getCallablesFromCallMap('function_exists')
#4 /var/htdocs/vendor/vimeo/psalm/src/Psalm/Internal/PhpVisitor/ReflectorVisitor.php(340): Psalm\Internal\PhpVisitor\ReflectorVisitor->registerClassMapFunctionCall('function_exists', Object(PhpParser\Node\Expr\FuncCall))
#5 /var/htdocs/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(123): Psalm\Internal\PhpVisitor\ReflectorVisitor->enterNode(Object(PhpParser\Node\Expr\FuncCall))
#6 /var/htdocs/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(146): PhpParser\NodeTraverser->traverseNode(Object(PhpParser\Node\Expr\BooleanNot))
#7 /var/htdocs/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(223): PhpParser\NodeTraverser->traverseNode(Object(PhpParser\Node\Stmt\If_))
#8 /var/htdocs/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(114): PhpParser\NodeTraverser->traverseArray(Array)
#9 /var/htdocs/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(223): PhpParser\NodeTraverser->traverseNode(Object(PhpParser\Node\Stmt\If_))
#10 /var/htdocs/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(91): PhpParser\NodeTraverser->traverseArray(Array)
#11 /var/htdocs/vendor/vimeo/psalm/src/Psalm/Internal/Scanner/FileScanner.php(94): PhpParser\NodeTraverser->traverse(Array)
#12 /var/htdocs/vendor/vimeo/psalm/src/Psalm/Internal/Codebase/Scanner.php(586): Psalm\Internal\Scanner\FileScanner->scan(Object(Psalm\Codebase), Object(Psalm\Storage\FileStorage), false, Object(Psalm\Progress\DefaultProgress))
#13 /var/htdocs/vendor/vimeo/psalm/src/Psalm/Internal/Codebase/Scanner.php(367): Psalm\Internal\Codebase\Scanner->scanFile('/var/htdocs/ven...', Array, true)
#14 /var/htdocs/vendor/vimeo/psalm/src/Psalm/Internal/Codebase/Scanner.php(475): Psalm\Internal\Codebase\Scanner->Psalm\Internal\Codebase\{closure}(0, '/var/htdocs/ven...')
#15 /var/htdocs/vendor/vimeo/psalm/src/Psalm/Internal/Codebase/Scanner.php(325): Psalm\Internal\Codebase\Scanner->scanFilePaths(1)
#16 /var/htdocs/vendor/vimeo/psalm/src/Psalm/Config.php(1864): Psalm\Internal\Codebase\Scanner->scanFiles(Object(Psalm\Internal\Codebase\ClassLikes))
#17 /var/htdocs/vendor/vimeo/psalm/src/Psalm/Internal/Analyzer/ProjectAnalyzer.php(312): Psalm\Config->visitComposerAutoloadFiles(Object(Psalm\Internal\Analyzer\ProjectAnalyzer), Object(Psalm\Progress\DefaultProgress))
#18 /var/htdocs/vendor/vimeo/psalm/src/Psalm/Internal/Analyzer/ProjectAnalyzer.php(497): Psalm\Internal\Analyzer\ProjectAnalyzer->visitAutoloadFiles()
#19 /var/htdocs/vendor/vimeo/psalm/src/psalm.php(585): Psalm\Internal\Analyzer\ProjectAnalyzer->check('/var/htdocs/', false)
#20 /var/htdocs/vendor/vimeo/psalm/psalm(2): require_once('/var/htdocs/ven...')
#21 {main}
(Psalm dev-master@067104e170da5b0cf4a3cebcac58ecf36764c1f9 crashed due to an uncaught Throwable)
```

**Comment by jrushlow** [comment]
> Hey @jrushlow, can you reproduce the issue on https://psalm.dev ?

no

**Comment by andrew-demb** [comment]
Your PHP version under 7.3?
Can you check https://github.com/vimeo/psalm/pull/3090 ?

**Comment by muglug** [comment]
Sorry, that should be fixed in 41b165c

**Comment by jrushlow** [comment]
PHP 7.2.29, ill check in just a sec
